typescript/sdk/src/deploy/verify/ContractVerifier.ts

```diff
@@ -7,167 +7,109 @@ import { sleep, strip0x } from '@hyperlane-xyz/utils';
 import { ExplorerFamily } from '../../metadata/chainMetadataTypes';
 import { MultiProvider } from '../../providers/MultiProvider';
 import { ChainMap, ChainName } from '../../types';
-import { MultiGeneric } from '../../utils/MultiGeneric';
 
 import {
   CompilerOptions,
   ContractVerificationInput,
-  VerificationInput,
+  ExplorerApiActions,
+  ExplorerApiErrors,
+  ExplorerLicenseType,
+  FormOptions,
 } from './types';
 
-enum ExplorerApiActions {
-  GETSOURCECODE = 'getsourcecode',
-  VERIFY_IMPLEMENTATION = 'verifysourcecode',
-  MARK_PROXY = 'verifyproxycontract',
-  CHECK_STATUS = 'checkverifystatus',
-  CHECK_PROXY_STATUS = 'checkproxyverification',
-}
-
-enum ExplorerApiErrors {
-  ALREADY_VERIFIED = 'Contract source code already verified',
-  ALREADY_VERIFIED_ALT = 'Already Verified',
-  VERIFICATION_PENDING = 'Pending in queue',
-  PROXY_FAILED = 'A corresponding implementation contract was unfortunately not detected for the proxy address.',
-  BYTECODE_MISMATCH = 'Fail - Unable to verify. Compiled contract deployment bytecode does NOT match the transaction deployment bytecode.',
-}
-
-export class ContractVerifier extends MultiGeneric<VerificationInput> {
-  protected logger: Debugger;
+export class ContractVerifier {
+  private logger = debug(`hyperlane:ContractVerifier`);
+  private compilerOptions: CompilerOptions;
 
   constructor(
-    verificationInputs: ChainMap<VerificationInput>,
-    protected readonly multiProvider: MultiProvider,
-    protected readonly apiKeys: ChainMap<string>,
-    protected readonly source: string, // solidity standard input json
-    protected readonly compilerOptions: CompilerOptions,
+    private multiProvider: MultiProvider,
+    private readonly apiKeys: ChainMap<string>,
+    private source: string, // solidity standard input json
+    compilerOptions: Partial<Omit<CompilerOptions, 'codeformat'>>,
   ) {
-    super(verificationInputs);
-    this.logger = debug('hyperlane:ContractVerifier');
-  }
-
-  verify(targets = this.chains()): Promise<PromiseSettledResult<void>[]> {
-    return Promise.allSettled(
-      targets.map((chain) => {
-        const { family } = this.multiProvider.getExplorerApi(chain);
-        if (family === ExplorerFamily.Other) {
-          this.logger(
-            `Skipping verification for ${chain} due to unsupported explorer family.`,
-          );
-          return Promise.resolve();
-        }
-        return this.verifyChain(chain, this.get(chain));
-      }),
-    );
-  }
-
-  async verifyChain(
-    chain: ChainName,
-    inputs: VerificationInput,
-  ): Promise<void> {
-    this.logger(`Verifying ${chain}...`);
-    for (const input of inputs) {
-      await this.verifyContract(chain, input);
-    }
+    this.compilerOptions = {
+      codeformat: 'solidity-standard-json-input',
+      compilerversion:
+        compilerOptions?.compilerversion ?? 'v0.8.19+commit.7dd6d404',
+      licenseType: compilerOptions?.licenseType ?? ExplorerLicenseType.MIT,
```

Comment:
jmrossy
> Nit, is it better to assume MIT or just have this field be optional?

Reply:
I had it fallback to MIT to minimise config but it's probably better to leave it blank as it's optional on explorer APIs